scipy/sparse/dia.py

```diff
@@ -72,6 +72,19 @@ class dia_matrix(_data_matrix):
            [0, 2, 3, 0],
            [0, 0, 3, 4]])
 
+    >>> n= 10000
+    >>> ex = np.ones(n)
+    >>> data = np.array([ex, 2 * ex, ex])
+    >>> offsets = np.array([-1, 0, 1])
+    >>> sp.sparse.dia_matrix((data, offsets), shape=(n, n)).toarray()
```

Comment:
CI error is real though: the example needs to import `dia_matrix` explicitly (but `import numpy as np` is implicit in all examples)

```
scipy.sparse.dia_matrix
-----------------------
File "build/testenv/lib/python3.6/site-packages/scipy/sparse/dia.py", line 79, in dia_matrix
Failed example:
    sp.sparse.dia_matrix((data, offsets), shape=(n, n)).toarray()
Exception raised:
    Traceback (most recent call last):
      File "/opt/python/3.6.7/lib/python3.6/doctest.py", line 1330, in __run
        compileflags, 1), test.globs)
      File "<doctest dia_matrix[10]>", line 1, in <module>
        sp.sparse.dia_matrix((data, offsets), shape=(n, n)).toarray()
    NameError: name 'sp' is not defined
```

Reply:
Thanks for point that out, I just fixed this problem. Maybe not that large, but I feel like many non-advanced users could benefit a bit from an example that shows how to create dia_matrix without explicitly write out all rows.

Reply:
What I'm saying is that the example needs an explicit `from scipy.sparse import dia_matrix`.

Re size I'm suggesting to use n=10 or so. 

The example is great regardless of the size.

Reply:
I see, I just made two editing based on your advice. I definitely agree that for example 10000 is too large. I just set n = 10. Thanks.

Reply:
@ev-br

Reply:
Well, the refguide-check still fails 
```
scipy.sparse.dia_matrix
1240-----------------------
1241
1242File "build/testenv/lib/python3.6/site-packages/scipy/sparse/dia.py", line 81, in dia_matrix
1243Failed example:
1244    dia_matrix((data, offsets), shape=(n, n)).toarray()
1245Expected nothing
1246Got:
1247    array([[2., 1., 0., 0., 0., 0., 0., 0., 0., 0.],
1248           [1., 2., 1., 0., 0., 0., 0., 0., 0., 0.],
1249           [0., 1., 2., 1., 0., 0., 0., 0., 0., 0.],
1250           [0., 0., 1., 2., 1., 0., 0., 0., 0., 0.],
1251           [0., 0., 0., 1., 2., 1., 0., 0., 0., 0.],
1252           [0., 0., 0., 0., 1., 2., 1., 0., 0., 0.],
1253           [0., 0., 0., 0., 0., 1., 2., 1., 0., 0.],
1254           [0., 0., 0., 0., 0., 0., 1., 2., 1., 0.],
1255           [0., 0., 0., 0., 0., 0., 0., 1., 2., 1.],
1256           [0., 0., 0., 0., 0., 0., 0., 0., 1., 2.]])
```

It'd be good if you check `$ python runtests.py -s sparse --refguide-check` locally.